Source/VOE/Outpost_Defensive.cs

```diff
@@ -36,51 +46,98 @@ static Outpost_Defensive()
                 if (map.Parent is AmbushedRaid ambushedRaid)
                 {
-                    ambushedRaid.DefensiveOutpost = defense;
+                    ambushedRaid.DefensiveOutpost = this;
                     ambushedRaid.raidFaction = parms.faction;
+                    ambushedRaid.raidPoints = parms.points;
                 }
-                var pawns = defense.AllPawns.Where(x=>!x.IsPrisonerOfColony).InRandomOrder().Skip(1).ToList();
-                var defaultPawnGroupMakerParms = IncidentParmsUtility.GetDefaultPawnGroupMakerParms(PawnGroupKindDefOf.Combat, parms, ensureCanGenerateAtLeastOnePawn: true);
                 defaultPawnGroupMakerParms.generateFightersOnly = true;
                 defaultPawnGroupMakerParms.dontUseSingleUseRocketLaunchers = true;
+
                 var enemies = PawnGroupMakerUtility.GeneratePawns(defaultPawnGroupMakerParms).ToList();
                 MultipleCaravansCellFinder.FindStartingCellsFor2Groups(map, out var playerSpot, out var enemySpot);
-                foreach (var pawn in pawns) GenSpawn.Spawn(defense.RemovePawn(pawn), CellFinder.RandomSpawnCellForPawnNear(playerSpot, map), map, Rot4.Random);
+                var pawns = AllPawns.Where(x => !x.IsPrisonerOfColony).InRandomOrder().Skip(1).ToList();
+                foreach (var pawn in pawns) GenSpawn.Spawn(RemovePawn(pawn), CellFinder.RandomSpawnCellForPawnNear(playerSpot, map), map, Rot4.Random);
                 foreach (var enemy in enemies) GenSpawn.Spawn(enemy, CellFinder.RandomSpawnCellForPawnNear(enemySpot, map), map, Rot4.Random);
                 var lordJob = new LordJob_AssaultColony(parms.faction, true, false);
                 LordMaker.MakeNewLord(parms.faction, lordJob, map, enemies);
                 var letterLabel = "Outposts.Letters.Intercept.Label".Translate();
-                var letterText = "Outposts.Letters.Intercept.Text".Translate(targetMap.Parent.LabelCap, defense.Name);
+                var letterText = "Outposts.Letters.Intercept.Text".Translate(targetMap.Parent.LabelCap, Name);
                 PawnRelationUtility.Notify_PawnsSeenByPlayer_Letter(enemies, ref letterLabel, ref letterText,
                     "LetterRelatedPawnsGroupGeneric".Translate(Faction.OfPlayer.def.pawnsPlural), true);
                 Find.LetterStack.ReceiveLetter(letterLabel, letterText, LetterDefOf.PositiveEvent,
-                    new LookTargets(new List<GlobalTargetInfo> {defense, map.Parent}));
+                    new LookTargets(new List<GlobalTargetInfo> { this, map.Parent }));
                 Find.TickManager.Notify_GeneratedPotentiallyHostileMap();
             }, "GeneratingMapForNewEncounter", false, null, false);
-            return false;
         }
-
+        private void DeclineIntercept(IncidentParms parms, IncidentWorker_RaidEnemy raid)
+        {
+            DoRaid = true;
+            if (raid.TryExecute(parms))
```

Comment:
Wouldn't this cause the outpost to get stuck never intercepting if for some reason this returned `false`?

Reply:
I suppose it could until reload. My concern was if I set it to false without knowing the raid fired it might cause a loop of dialogs. Though, if that is an issue there would also be an issue of intercepting raids that actually never would've gotten to the colony. 

I don't think there is an issue but we can add least add one safe guard for confirming generate Faction worked. That looks like the only thing that would return false in Execute for RaidEnemy. There is another if 0 pawns which I can add a check for before we get to dialog. But its not something I think ever really comes up, since it'd also log an error.

Reply:
Sounds like a plan.